src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java

```diff
@@ -0,0 +1,390 @@
+		System.out.println("taskbarNativeHeight: " + taskbarNativeHeight);
+		System.out.println("taskbarNativeWidth: " + taskbarNativeWidth);
+		System.out.println("taskbarWebviewHeight: " + taskbarWebviewHeight);
+		System.out.println("taskbarWebviewWidth: " + taskbarWebviewWidth);
+		System.out.println("=======================================");
+		System.out.println("appNativeWidth: " + appNativeWidth);
+		System.out.println("appNativeHeight: " + appNativeHeight);
+		System.out.println("appWebviewHeight: " + appWebviewHeight);
+		System.out.println("appWebviewWidth: " + appWebviewWidth);
+		System.out.println("=======================================");
+		System.out.println("loadedPageHeight: " + loadedPageHeight);
+		System.out.println("loadedPageWidth: " + loadedPageWidth);
+		System.out.println("=======================================");
+	}
+	
+	/**
+	 * 
+	 * @param direction Use public const DIRECTION from that class
+	 * @param pixelPath	From 0 to half of screen X or Y, if path will be too high it will be set to edge of app
+	 * @param duration In milliseconds
+	 * @param waitAfter In milliseconds, Reccomend 2*duration
+	 */
+	public void swipeFromCenterToDirection(String direction, int pixelPath, int duration, int waitAfter){
+		int centerX = appNativeWidth / 2;
+		int centerY = (appNativeHeight / 2) + taskbarNativeHeight;
+		int path = 0;
+		if (!"NATIVE_APP".equals(mobileDriver.getContext())) {
+			mobileDriver.context("NATIVE_APP");
+		}
+		switch (direction) {
+			case "left": 
+				if (pixelPath < centerX) {
+					path = pixelPath;
+				} else {
+					path = centerX - 2;
+				}
+				try {
+					mobileDriver.swipe(centerX, centerY, centerX - path, centerY, duration);
+				} catch (Exception e) {
+					PageObjectLogging.log("swipeFromCenterToDirection", e.toString(), false);
+				}
+				break;
+			case "right": 
+				if (pixelPath < centerX) {
+					path = pixelPath;
+				} else {
+					path = centerX - 2;
+				}
+				try {
+					mobileDriver.swipe(centerX, centerY, centerX + path, centerY, duration);	
+				} catch (Exception e) {
+					PageObjectLogging.log("swipeFromCenterToDirection", e.toString(), false);
+				}
+				break;
+			case "up": 
+				if (pixelPath < centerY) {
+					path = pixelPath;
+				} else {
+					path = centerY - 2;
+				}
+				try {
+					mobileDriver.swipe(centerX, centerY, centerX, centerY - path, duration);				
+				} catch (Exception e) {
+					PageObjectLogging.log("swipeFromCenterToDirection", e.toString(), false);
+				}
+				break;
+			case "down": 
+				if (pixelPath < centerY) {
+					path = pixelPath;
+				} else {
+					path = centerY - 2;
+				}
+				try {						
+					mobileDriver.swipe(centerX, centerY, centerX, centerY + path, duration);	
+				} catch (Exception e) {
+					PageObjectLogging.log("swipeFromCenterToDirection", e.toString(), false);
+				}
+				break;
+			default: break;
+		}
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {
```

Comment:
Please specified which exception the try block is catching rather than a generic exception.